keel-web/src/test/kotlin/com/netflix/spinnaker/keel/rest/AuthorizationSupportTests.kt

```diff
@@ -0,0 +1,233 @@
+            ).isFalse()
+          }
+        }
+      }
+
+      listOf(Action.READ, Action.WRITE).forEach { action ->
+        context("user has no ${action.name} access to cloud account") {
+          before {
+            every {
+              permissionEvaluator.hasPermission(any() as Authentication, any(), "ACCOUNT", action.name)
+            } returns false
+          }
+
+          test("permission check specifying application name fails") {
+            expectThrows<AccessDeniedException> {
+              checkCloudAccountPermission(action, TargetEntity.APPLICATION, application)
+            }
+            expectThat(
+              hasCloudAccountPermission(action.name, TargetEntity.APPLICATION.name, application)
+            ).isFalse()
+          }
+
+          test("permission check specifying resource id fails") {
+            expectThrows<AccessDeniedException> {
+              checkCloudAccountPermission(action, TargetEntity.RESOURCE, resource.id)
+            }
+            expectThat(
+              hasCloudAccountPermission(action.name, TargetEntity.RESOURCE.name, resource.id)
+            ).isFalse()
+          }
+
+          test("permission check specifying delivery config name fails") {
+            expectThrows<AccessDeniedException> {
+              checkCloudAccountPermission(action, TargetEntity.DELIVERY_CONFIG, deliveryConfig.name)
+            }
+            expectThat(
+              hasCloudAccountPermission(action.name, TargetEntity.DELIVERY_CONFIG.name, deliveryConfig.name)
+            ).isFalse()
+          }
+        }
+      }
+
+      context("user has no access to service account") {
+        before {
+          every {
+            permissionEvaluator.hasPermission(any() as Authentication, any(), "SERVICE_ACCOUNT", any())
+          } returns false
+        }
+
+        test("permission check specifying application name fails") {
+          expectThrows<AccessDeniedException> {
+            checkServiceAccountAccess(TargetEntity.APPLICATION, application)
+          }
+          expectThat(
+            hasServiceAccountAccess(TargetEntity.APPLICATION.name, application)
+          ).isFalse()
+        }
+
+        test("permission check specifying resource id fails") {
+          expectThrows<AccessDeniedException> {
+            checkServiceAccountAccess(TargetEntity.RESOURCE, resource.id)
+          }
+          expectThat(
+            hasServiceAccountAccess(TargetEntity.RESOURCE.name, resource.id)
+          ).isFalse()
+        }
+
+        test("permission check specifying delivery config name fails") {
+          expectThrows<AccessDeniedException> {
+            checkServiceAccountAccess(TargetEntity.DELIVERY_CONFIG, deliveryConfig.name)
+          }
+          expectThat(
+            hasServiceAccountAccess(TargetEntity.DELIVERY_CONFIG.name, deliveryConfig.name)
+          ).isFalse()
+        }
+      }
+    }
+
+    context("authorization is disabled") {
```

Comment:
Are there any tests were authorization is enabled and the user has the required permissions? I didn't see any.

Reply:
Never mind, I see these in the Controller tests.